source/clear-linux/tutorials/hadoop.rst

```diff
@@ -0,0 +1,177 @@
+.. _hadoop:
+
+Set up a single node cluster with Hadoop\*
+##########################################
+
+This tutorial walks you through the process of installing, configuring, and
+running Apache\* Hadoop on |CLOSIA|. The Apache Hadoop software library is a
+framework for distributed processing of large data sets across clusters of
+computers using simple programming models. It is designed to scale up from
+single servers to thousands of machines, with each machine offering local
+computation and storage.
+
+Prerequisites
+*************
+
+Before following this tutorial, you should follow the
+:ref:`bare-metal-install` to ensure you have installed |CLOSIA|.
+
+Before you install any new packages, update |CL| with the following command:
+
+.. code-block:: bash
+
+   sudo swupd update
+
+For the purposes of this tutorial, we will install Hadoop in a single machine
+running both the master and slave daemons.
+
+Install Apache Hadoop
+*********************
+
+Apache Hadoop is included in the `big-data-basic` bundle. To install the
+framework, enter the following command:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add big-data-basic
+
+Configure Apache Hadoop
+***********************
+
+#. To create the configuration directory, enter the following command:
+
+   .. code-block:: bash
+
+      sudo mkdir /etc/hadoop
+
+#. Copy the defaults from :file:`/usr/share/defaults/hadoop` to
+   :file:`/etc/hadoop` with the following command:
+
+   .. code-block:: bash
+
+      $ sudo cp /usr/share/defaults/hadoop/* /etc/hadoop
+
+   .. note:: Remember, you should never modify the files under the
+      :file:`/usr/share/defaults` directory since swupd will overwrite them.
+
+Once all configuration files are in :file:`/etc/hadoop`, we must edit them
+according to our needs. The first file, :file:`/etc/hadoop/core-site.xml`, is
+responsible to inform the Hadoop daemon of where `NameNode` running. The
+`NameNode` server is the master server managing the files system namespace
+and regulating the clients' access to files.
+
+In this tutorial, our `NameNode` runs in our `localhost`. Follow these steps
+to set it up correctly:
+
+#. Open the :file:`/etc/hadoop/core-site.xml` using the editor of your
+   choice and modify the file as follows:
+
+   .. code-block:: xml
+
+      <?xml version="1.0" encoding="UTF-8"?>
+      <?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
+      <configuration>
+      <property>
+      <name>fs.default.name</name>
+      <value>hdfs://localhost:9000</value>
+      </property>
+      </configuration>
+
+#. Edit :file:`/etc/hadoop/hdfs-site.xml`. This file configures the
+   :abbr:`HDFS (Hadoop Distributed File System)` daemons. This includes
+   things like the list of permitted and excluded data nodes or the size of
+   the blocks. In this example, we are setting the number of block
+   replication to 1 from the default of 3 as follows:
+
+   .. code-block:: xml
```

Comment:
Something is b0rky here and the xml code cannot be rendered correctly in github, dunno if it would be the case one the rst is rendered to the webpage.
@rcaballeromx mind to confirm this is buggy?

Reply:
Yes, GitHub does not render code blocks with line emphasis, luckily, Sphinx does. I have tested the content by building the html locally and these render correctly. Sadly, the current website CSS does not support syntax highlighting, but the new one will.